pkg/policy/k8s/cell.go

```diff
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package k8s
+
+import (
+	"context"
+
+	"github.com/cilium/cilium/pkg/hive"
+	"github.com/cilium/cilium/pkg/hive/cell"
+	"github.com/cilium/cilium/pkg/k8s"
+	cilium_v2 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2"
+	cilium_v2_alpha1 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1"
+	"github.com/cilium/cilium/pkg/k8s/client"
+	"github.com/cilium/cilium/pkg/k8s/resource"
+	slim_networking_v1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/api/networking/v1"
+	"github.com/cilium/cilium/pkg/k8s/synced"
+	"github.com/cilium/cilium/pkg/labels"
+	"github.com/cilium/cilium/pkg/logging"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/policy"
+	"github.com/cilium/cilium/pkg/policy/api"
+	"github.com/cilium/cilium/pkg/promise"
+)
+
+const (
+	k8sAPIGroupNetworkingV1Core                 = "networking.k8s.io/v1::NetworkPolicy"
+	k8sAPIGroupCiliumNetworkPolicyV2            = "cilium/v2::CiliumNetworkPolicy"
+	k8sAPIGroupCiliumClusterwideNetworkPolicyV2 = "cilium/v2::CiliumClusterwideNetworkPolicy"
+	k8sAPIGroupCiliumCIDRGroupV2Alpha1          = "cilium/v2alpha1::CiliumCIDRGroup"
+)
+
+var Cell = cell.Module(
+	"policy-k8s-watcher",
+	"Watches K8s policy related objects",
+
+	cell.Invoke(startK8sPolicyWatcher),
+)
+
+var (
+	log = logging.DefaultLogger.WithField(logfields.LogSubsys, "k8s-policy-watcher")
```

Comment:
I'm understanding you left the log as is to not make this PR bigger?

Note that the logging is calling it `k8s-policy-watcher` whereas the module is called `policy-k8s-watcher`. If you intend to transform the logging at some point, that module name will become the subsystem name, which would hence change. Are we fine with that?

Reply:
> I'm understanding you left the log as is to not make this PR bigger?

Yes, though I think I might as well transform that too if the diff is not too big (since follow-up PRs are likely going to focus on different areas of the code).

Reply:
Pushed an additional commit which uses the hive based logging.